frontend/src/app/supportAdmin/ManageFacility/ManageFacility.tsx

```diff
@@ -0,0 +1,185 @@
+  facility: undefined,
+};
+
+const ManageFacility = () => {
+  useDocumentTitle("Manage Facility");
+  const [localState, updateLocalState] =
+    useState<ManageFacilityState>(initialState);
+
+  /**
+   * Fetch organizations (on initial load)
+   */
+  const { data: orgResponse, loading: loadingOrgs } =
+    useGetAllOrganizationsQuery();
+
+  const orgOptions: Option[] =
+    orgResponse?.organizations?.map((org) => ({
+      value: org.id,
+      label: org.name,
+    })) ?? [];
+
+  /**
+   * Fetch facilities
+   */
+  const [
+    queryGetFacilitiesByOrgId,
+    { data: facilitiesResponse, loading: loadingFacilities },
+  ] = useGetFacilitiesByOrgIdLazyQuery();
+
+  const facilitiesOptions: Option[] =
+    facilitiesResponse?.organization?.facilities.map((facility) => ({
+      value: facility.id,
+      label: facility.name,
+    })) ?? [];
+
+  /**
+   * Fetch facility stats
+   */
+  const [queryGetFacilityStats] = useGetFacilityStatsLazyQuery();
+
+  /**
+   * Facility select filter handlers
+   */
+  function handleSelectOrganization(e: React.ChangeEvent<HTMLSelectElement>) {
+    const selectedOrg = e.target.value;
+    if (selectedOrg !== "") {
+      queryGetFacilitiesByOrgId({
+        variables: { orgId: selectedOrg },
+        fetchPolicy: "no-cache",
+      }).then();
+    }
+
+    updateLocalState((prevState) => ({
+      ...prevState,
+      orgId: selectedOrg,
+      facilityId: "",
+    }));
+  }
+  async function handleSelectFacility(e: React.ChangeEvent<HTMLSelectElement>) {
+    const facilityId = e.target.value;
+
+    updateLocalState((prevState) => ({
+      ...prevState,
+      facilityId: facilityId,
+    }));
+  }
+
+  async function handleSearch() {
+    const facilityId = localState.facilityId;
+
+    if (facilityId === "") {
+      updateLocalState((prevState) => ({
+        ...prevState,
+        facility: undefined,
+      }));
+    } else {
+      const selectedFacility =
+        facilitiesResponse?.organization?.facilities?.filter(
+          (f) => f.id === facilityId
+        )?.[0];
+
+      const facilityStats = await queryGetFacilityStats({
+        fetchPolicy: "no-cache",
+        variables: { facilityId },
+      }).then((response) => response.data?.facilityStats);
+
+      updateLocalState((prevState) => ({
+        orgId: prevState.orgId,
+        facilityId: facilityId,
+        facility: {
+          city: selectedFacility?.city || "",
+          state: selectedFacility?.state || "",
+          zipcode: selectedFacility?.zipCode || "",
+          id: selectedFacility?.id || "",
+          name: selectedFacility?.name || "",
+          org: facilitiesResponse?.organization?.name || "",
+          orgType: facilitiesResponse?.organization?.type || "",
+          usersCount: facilityStats?.usersSingleAccessCount || 0,
+          patientsCount: facilityStats?.patientsSingleAccessCount || 0,
+        },
+      }));
+    }
+  }
+
+  function handleClearFilter() {
+    updateLocalState(initialState);
```

Comment:
just so you know -- this does not clear the clear the facilities list.
to repro:
- select an org with facilities
- check the facilities dropdown and see there are facilities
- click "clear filters"
- see the org is unselected but facilities list remains

Not sure if you want to do this as well but I passed facilities as part of the localState object and cleared it as part of this method

Reply:
Fixed!